be/src/vec/columns/column.h

```diff
@@ -411,10 +413,15 @@ class IColumn : public COW<IColumn> {
     // only used in ColumnNullable replace_column_data
     virtual void replace_column_data_default(size_t self_row = 0) = 0;
 
-    bool is_date_type() { return is_date; }
+    virtual bool is_date_type() { return is_date; }
```

Comment:
is_xxx should be marked as const

Reply:
This is a temporary implementation, this method would be removed later.